src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,226 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+export type PaginationProps = PropsWithChildren<{
+  className?: string;
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  // Optional props
+  showIcon?: boolean;
+  showIconOnly?: boolean;
+  showOnlyPreviousAndNext?: boolean;
+  tableDataPagination?: boolean;
+}>;
+
+export const Pagination: FC<PaginationProps> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+  showIcon,
+  showIconOnly,
+  showOnlyPreviousAndNext,
+  tableDataPagination,
+}) => {
+  // Logic for displaying page numbers
+  const pageNumbers = [];
+  const startPage = Math.max(1, currentPage - 3);
+  const endPage = Math.min(currentPage + 3, totalPages);
+  for (let i = startPage; i <= endPage; i++) {
```

Comment:
I don't think that we need to run this loop for all the pagination styles. Do we?

Reply:
This is for generating only few page numbers array based on the current page @rluders. 
For example:
current page =5, it will only display 2 3 4 5 6 7 8 page numbers for selection.

Reply:
Yeah, I got it.

But, it will only make sense if you want to show those numbers, right?
If you are just showing the `Next` and `Prev` buttons... This loop doesn't make any sense... or did I got it wrong?

Reply:
ooh yeah. got it @rluders.  Thanks

Reply:
done